pkg/context/project.go

```diff
@@ -0,0 +1,93 @@
+package context
+
+import (
+	"errors"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+	"io/ioutil"
+	"os"
+	"path"
+	"strmprivacy/strm/pkg/common"
+	"strmprivacy/strm/pkg/entity/project"
+)
+
+const defaultProjectFilename = "default_project"
+
+// ResolveProject resolves the project to use and makes its ID globally available.
+// The value passed through the flag takes precedence, then the value stored in the config dir, and finally
+// a fallback to default project.
+func ResolveProject(f *pflag.FlagSet) {
+
+	activeProjectFilepath := path.Join(common.ConfigPath, defaultProjectFilename)
+	projectFlagValue, _ := f.GetString(ProjectFlag)
+
+	if _, err := os.Stat(activeProjectFilepath); (os.IsNotExist(err) || GetDefaultProject() == "") && projectFlagValue == "" {
+		initDefaultProject()
+	}
+
+	if projectFlagValue != "" {
+		activeProject := project.GetProject(projectFlagValue)
+		if activeProject == nil {
+			message := fmt.Sprintf("Specified project '%v' does not exist, or you do not have access to it.", projectFlagValue)
+			common.CliExit(errors.New(message))
+		}
+		common.ProjectId = activeProject.Id
+	} else {
+		activeProject := project.GetProject(GetDefaultProject())
+		if activeProject == nil {
+			message := fmt.Sprintf("Default project '%v' does not exist, or you do not have access to it. " +
+				"A new default we be set.", GetDefaultProject())
+			fmt.Println(message)
```

Comment:
This can be a `common.CliExit(errors.New(message))`, and you can move the `initDefaultProject` above that line.

Reply:
That is what I had initially, but the reason I did it like this is because the `initDefaultProject` does a `fmt.Println` and when using `CliExit`, the order of the messages would be inverted. But I can move the print from `initDefaultProject` to outside that method, that will result in a single, clearer message 👍

Reply:
Ah ok, that does make sense indeed. Either ways would work, so it's your preference then.